string location
string task
---
string result
